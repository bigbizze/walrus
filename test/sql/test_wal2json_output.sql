select 1 from pg_create_logical_replication_slot('realtime', 'wal2json', false);

create table public.notes(
    id int primary key,
    body text
);

begin;
    insert into public.notes(id, body) values (1, 'hello');
    update public.notes set body = 'world';
    delete from public.notes;
end;

select
    jsonb_pretty(norm(x.data::jsonb))
from
    pg_logical_slot_get_changes(
        'realtime', null, null,
        'include-pk', '1',
        'include-transaction', 'false',
        'include-timestamp', 'true',
        'include-type-oids', 'true',
        'format-version', '2',
        'actions', 'insert,update,delete',
        'add-tables', 'public.notes'
    ) x;

select pg_drop_replication_slot('realtime');
drop table public.notes;
truncate table realtime.subscription;
