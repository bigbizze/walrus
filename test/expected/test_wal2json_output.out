select 1 from pg_create_logical_replication_slot('realtime', 'wal2json', false);
 ?column? 
----------
        1
(1 row)

create table public.notes(
    id int primary key,
    body text
);
begin;
    insert into public.notes(id, body) values (1, 'hello');
    update public.notes set body = 'world';
    delete from public.notes;
end;
select
    norm(x.data::jsonb)
from
    pg_logical_slot_get_changes(
        'realtime', null, null,
        'include-pk', '1',
        'include-transaction', 'false',
        'include-timestamp', 'true',
        'include-type-oids', 'true',
        'format-version', '2',
        'actions', 'insert,update,delete',
        'add-tables', 'public.notes'
    ) x;
                       norm                       
--------------------------------------------------
 {                                               +
     "pk": [                                     +
         {                                       +
             "name": "id",                       +
             "type": "integer",                  +
             "typeoid": 23                       +
         }                                       +
     ],                                          +
     "table": "notes",                           +
     "action": "I",                              +
     "schema": "public",                         +
     "columns": [                                +
         {                                       +
             "name": "id",                       +
             "type": "integer",                  +
             "value": 1,                         +
             "typeoid": 23                       +
         },                                      +
         {                                       +
             "name": "body",                     +
             "type": "text",                     +
             "value": "hello",                   +
             "typeoid": 25                       +
         }                                       +
     ],                                          +
     "timestamp": "2000-01-01 01:01:01.000000-07"+
 }
 {                                               +
     "pk": [                                     +
         {                                       +
             "name": "id",                       +
             "type": "integer",                  +
             "typeoid": 23                       +
         }                                       +
     ],                                          +
     "table": "notes",                           +
     "action": "U",                              +
     "schema": "public",                         +
     "columns": [                                +
         {                                       +
             "name": "id",                       +
             "type": "integer",                  +
             "value": 1,                         +
             "typeoid": 23                       +
         },                                      +
         {                                       +
             "name": "body",                     +
             "type": "text",                     +
             "value": "world",                   +
             "typeoid": 25                       +
         }                                       +
     ],                                          +
     "identity": [                               +
         {                                       +
             "name": "id",                       +
             "type": "integer",                  +
             "value": 1,                         +
             "typeoid": 23                       +
         }                                       +
     ],                                          +
     "timestamp": "2000-01-01 01:01:01.000000-07"+
 }
 {                                               +
     "pk": [                                     +
         {                                       +
             "name": "id",                       +
             "type": "integer",                  +
             "typeoid": 23                       +
         }                                       +
     ],                                          +
     "table": "notes",                           +
     "action": "D",                              +
     "schema": "public",                         +
     "identity": [                               +
         {                                       +
             "name": "id",                       +
             "type": "integer",                  +
             "value": 1,                         +
             "typeoid": 23                       +
         }                                       +
     ],                                          +
     "timestamp": "2000-01-01 01:01:01.000000-07"+
 }
(3 rows)

select pg_drop_replication_slot('realtime');
 pg_drop_replication_slot 
--------------------------
 
(1 row)

drop table public.notes;
truncate table realtime.subscription;
